src/MICore/Transports/LocalTransport.cs

```diff
@@ -22,10 +22,11 @@ public override void InitStreams(LaunchOptions options, out StreamReader reader,
         {
             LocalLaunchOptions localOptions = (LocalLaunchOptions)options;
             string miDebuggerDir = System.IO.Path.GetDirectoryName(localOptions.MIDebuggerPath);
+            string miDebuggerArgs = localOptions.GetMiDebuggerArgs();
 
             Process proc = new Process();
             proc.StartInfo.FileName = localOptions.MIDebuggerPath;
-            proc.StartInfo.Arguments = "--interpreter=mi";
+            proc.StartInfo.Arguments = miDebuggerArgs;
```

Comment:
We format this string in the PipeLaunchOptions path and the UnixShellPortOptions too. I think it would be useful in all these cases.

Reply:
Ah. I see we are inconsistent between the json and Xml handling of the pipe launch options. In the Json case, where we do manipulate the gdb command line explicitly, we should probably add in the new args too.

Reply:
PipeLaunchOptions can modify the gdb/lldb command via pipeArguments.

UnixShellPort does not have a way to get the miDebuggerArgs since it is not a LocalLaunchOption.

Reply:
right, so wouldn't the args field would need to be added to the "AttachOptions" and then merged into the construction of the LocalLaunchOptions or UnixShellPortLaunchOptions in CreateForAttachRequest?

I guess we have ways of working around the absence of a specific arg in the pipe case.

Reply:
Wouldn't it only be for the LocalLaunchOptions since the miDebuggerArgs only exist if there if isServerMode is true?